public/calculateSol.js

```diff
@@ -0,0 +1,34 @@
+const earthDayInMilliseconds = 24 * 60 * 60 * 1000;
+const solDayInMilliseconds =
+  earthDayInMilliseconds + 39 * 60 * 1000 + 35.244 * 1000;
+
+function calculateSol() {
+  const convertSol = solDayInMilliseconds / earthDayInMilliseconds;
+  const marsRoverLandingDate = new Date(2012, 7, 6, 5, 17, 57);
+  const timeSinceLandingInMilliseconds = Date.now() - marsRoverLandingDate;
+  const earthDaysSinceMarsRoverLanded =
+    timeSinceLandingInMilliseconds / earthDayInMilliseconds;
+  const marsSols = Math.floor(earthDaysSinceMarsRoverLanded / convertSol);
+
+  document
+    .querySelector("#mars-sol")
+    .append(`Current Mars Sols of the Mars Rover: ${marsSols}`);
+}
+
+calculateSol();
+
+function displayEarthDaysSinceMarsRoverLanded() {
+  const millisecondsSinceMarsRoverLanded =
+    Date.now() - new Date(2012, 7, 6, 5, 17, 57).getTime();
```

Comment:
Did you know you can just subtract dates and achieve the same thing?

`new Date() - new Date(2012, 7, 6, 5, 17, 57)`

(It's because the subtraction operator _coerces_ the dates into their values, which is the internal milliseconds since the start of time.)

Reply:
Ooh thanks! :) I'll take a look at changing that now